### PR TITLE
rsx: Fix error context dma #2243

### PR DESCRIPTION
**Pull request by Zangetsu38**
Fix Issue https://github.com/RPCS3/rpcs3/issues/2243 Thx @kd-11 for help

### CODE REVIEW

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 1]
```diff
@@ -287,6 +287,7 @@ namespace rsx
 			{
 			case blit_engine::context_dma::to_memory_get_report: location = CELL_GCM_LOCATION_LOCAL; break;
 			case blit_engine::context_dma::report_location_main: location = CELL_GCM_LOCATION_MAIN; break;
+			case blit_engine::context_dma::memory_host_buffer: location = CELL_GCM_LOCATION_MAIN; break;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: You can avoid code duplication by letting the first case fall through
```
case blit_engine::context_dma::report_location_main:
case blit_engine::context_dma::memory_host_buffer:
    location = CELL_GCM_LOCATION_MAIN; break;
```

Comment (Zangetsu38) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: oh yes, its true ^^

**Comment by kd-11** [comment]
Looks okay. If we can confirm that the changeset actually fixes the affected games it should be good. If not, more work may be required.

**Comment by Zangetsu38** [comment]
@kd-11  I think more work render
Formula one have this 
F {rsx::thread} struct gsl::fail_fast thrown: GSL: Precondition failure at c:\rpcs3\3rdparty\gsl\include\span.h: 1432
But Killzone 2 use also this, and don't have this, i think depend game

**Comment by kd-11** [comment]
The championship error may be unrelated to this one. We'll need to confirm it wont break things silently, otherwise we have to add a warning so that if a game fails, we know what caused it. Can someone test killzone 2?

**Comment by Zangetsu38** [comment]
Yes, killzone 2 demo show error in actually master build, 
>F {rsx::thread} class std::runtime_error thrown: Unknown context dma 0xfeed0001
(in file C:\rpcs3\rpcs3\Emu\RSX\gcm_enums.cpp:996)

And with this commit show Black screen more log succes and this fatal
>F {SPU[0x2000005] Thread (LowPriorityCellSpursKernel0)} class std::runtime_error thrown: Unimplemented instruction
(in file C:\rpcs3\rpcs3\Emu\Cell\SPUASMJITRecompiler.cpp:920)

Press resume no change, so probleme caused by spu, and have tested several game, no caused probleme

**Comment by kd-11** [comment]
@Zangetsu38 Rebase for merge and I'll commit it.

**Comment by Zangetsu38** [comment]
yes is done :)

**Comment by SakataGintokiYT** [comment]
Nice ^_^

**Pull request merged**